### ISSUE TITLE
Improve search

### ISSUE THREAD

**Issue by bhousel** [openstreetmap/iD#5214]
### Hint text
iD's search box is currently too mysterious.  Let's add some hint text below it to help the user know what to use it for.

<img width="396" alt="screenshot 2018-08-10 10 28 18" src="https://user-images.githubusercontent.com/38784/43963387-33b003a0-9c88-11e8-8135-9a0b408497ea.png">

### Beyond that
I'd also like to think about what other kinds of things we can add here. 
For example Plus Codes are [currently trendy](https://github.com/openstreetmap/openstreetmap-website/pull/1818#issuecomment-412023682)

This is one of those weird situations where we actually aren't using a big block of UI for anything, so adding to it costs us almost nothing and can be done in a way that improves the UX.  (usually, adding features degrades UX). 

I'd be ok with supporting more in this box (like plus code, geohash, wikidata ids, even what.three.words) if it helps someone find a spot of earth in OSM and add stuff to it.

**Comment by manfredbrandl** [comment]
The hint would help. I do search for features by Name, maybe by type. It is also helpful that coordinates (or Plus Codes) and OSM-Object-IDs (Nodes, ways (Areas), Relations oder Notes) can be enternd. For me the later ones are not „searched“ for because they are an unmistakeable address to go to, Bit i do Not have a better Word than search for them. OSM-Object-IDs will require an Explantation. My proposal:
Here you can search for features (by name like California, Manhattan ot White House or by type like Cafe or Rwstaurant). You can also enter coordinates (e.g. lat/lng „39.12345, -75.987654“ to DMS „...“) or PlusCodes. You can also enter ids of OSM-Objects (e.g. n1234567 for a Node or w9876543 for a way/area or xxx for a Note).

**Comment by slhh** [comment]
Entering a coordinate should not only offer an option to jump to that coordinate, but also offer an option to place a node there.

Supporting a suffix to define zoom level or viewport size might also be useful, e.g.:
```
Manhatten @z10
39.12345, -75.987654 @2km
```
The user should also be able to enter a bounding box to define the viewport. Standard bounding box specification formats need to be supported to allow copy and paste from other applications.

The search box might also be used to enter some global commands using a special prefix, e.g.:
```
@@@setmode indoor
@@@setlanguage en
```
The commands might be documented in an advanced user manual only. This way we can add functionality without cluttering the user interface.

**Comment by bhousel** [comment]
> The commands might be documented in an advanced user manual only. This way we can add functionality without cluttering the user interface.

Hey @slhh - I actually do want to add something like this to iD for advanced stuff like #2929, but I don't want it in the search box.

**Comment by TAQ2** [comment]
Based on the screenshot of V3 search box in https://github.com/openstreetmap/iD/issues/6928#issuecomment-540212142 the UI has changed leaving no whitespace. 

Does this mean that you still want feature done on version 2? I'm new to the project so I don't know how much you are prioritising between the different versions

**Comment by leifr-unc** [comment]
Perhaps a drop-down with the hints could appear once that search bar is selected, and then be converted to an auto complete list once text is entered.  That would allow this feature to fit into v3.

**Comment by BjornRasmussen** [comment]
Perhaps a drop-down with the hints could appear once that search bar is selected, and then be converted to an auto complete list once text is entered.  That would allow this feature to fit into v3.

**Comment by quincylvania** [comment]
@TAQ2 Yeah this UI has been going through a lot of churn so I'd say let's come back to this issue later on.

**Comment by TAQ2** [comment]
@quincylvania

Please can you explain further what you mean by churn. 

Not trying to push the feature by any means but it seems like a straightforward task that would not require much code for improved UX.


I'm just asking because I would like more insight into where the project is going and what is being prioritised.

**Comment by quincylvania** [comment]
> Please can you explain further what you mean by churn.

@TAQ2 Sure. I mean that the feature search UI has changed a lot for v3 and may continue to change before release. While we could add this just for v2, I think our time would be better spent on stuff that improves both branches or the v3 branch.

**Comment by thibaultmol** [comment]
I only just now discovered that finding a way or other object requires you to put 'w' or some other letter in front of it. Maybe having some text when no results are returned and it's just a number that was entered say: "Specify your search term by adding 'w' in front of it in case of a way. See this link for other search operators..." or something

**Comment by thibaultmol** [comment]
Also, being able to just paste an osm link and have it realize what it means.
For example "https://www.openstreetmap.org/way/323660491" should just show that way

**Comment by quincylvania** [comment]
@thibaultmol That sounds okay but would be a separate issue.
